doc/branch.rst

```diff
@@ -129,3 +127,112 @@ While we might expect this line of code to be reported as covered, the
 generator did not iterate until ``StopIteration`` is raised, the indication
 that the loop is complete. This is another case
 where adding ``# pragma: no branch`` may be desirable.
+
+Explanations and examples of missing branches
+---------------------------------------------
+
+This section shows common examples of missing branches in Python code and
+explains how coverage.py reports them.
+
+Example::
+
+    10: for x in items:
+    11:     print(x)
+    12:     print("loop done")
+    13: print("done")
+
+Possible branches:
+
+* ``10 -> 11`` (enter loop body)
+* ``12 -> 10`` (repeat)
+* ``10 -> 13`` (skip loop entirely)
+
+
+Case 1 — ``items`` is empty::
+
+    Missing:
+        10 -> 11
+        12 -> 10
+
+Case 2 — ``items = [1]``::
+
+    Missing:
+        12 -> 10
```

Comment:
I'm sorry i didn't realize this earlier: this code doesn't have a branch from 12->10, because branches only occur where a statement can go next to two different places.  Line 12 always goes to line 10, even on the last iteration of the loop.  So we need to re-think this example.

Reply:
It's important to test the example to see that coverage.py actually reports the missing branches we're claiming.